jcl/src/java.base/share/classes/java/lang/StringBuffer.java

```diff
@@ -529,8 +529,86 @@ public StringBuffer append (float value) {
  * @param		value	the integer
  * @return		this StringBuffer
  */
-public StringBuffer append (int value) {
+public synchronized StringBuffer append(int value) {
+	/*[IF Sidecar19-SE]*/
+	/*[IF Sidecar19-SE-B174]*/
+	int currentLength = lengthInternalUnsynchronized();
+	int currentCapacity = capacityInternal();
+
+	int valueLength = Integer.stringSize(value);
+
+	int newLength = currentLength + valueLength;
+
+	if (newLength > currentCapacity) {
+		ensureCapacityImpl(newLength);
+	}
+
+	if (String.enableCompression && count >= 0) {
+		if (count >= 0) {
+			Integer.getChars(value, newLength, this.value);
+
+			count = newLength;
+		} else {
+			StringUTF16.getChars(value, newLength, this.value);
+
+			count = newLength | uncompressedBit;
+		}
+	} else {
+		StringUTF16.getChars(value, newLength, this.value);
+
+		count = newLength;
+	}
+
+	return this;
+	/*[ELSE]*/
 	return append(Integer.toString(value));
+	/*[ENDIF]*/
+	/*[ELSE]*/
+	if (value != Integer.MIN_VALUE) {
```

Comment:
There's basically two versions of the algorithm here with a single difference.  Rewriting as:

```
	/*[ELSE]*/
	if (value != Integer.MIN_VALUE) {
		if (String.enableCompression && count >= 0) {
			return append(Integer.toString(value));
		} else {
			int currentLength = lengthInternalUnsynchronized();
			int currentCapacity = capacityInternal();

			int valueLength = value < 0 ? Integer.stringSize(-value) + 1 : Integer.stringSize(value);

			int newLength = currentLength + valueLength;

			if (newLength > currentCapacity) {
				ensureCapacityImpl(newLength);
			}

			Integer.getChars(value, newLength, this.value);


			if (String.enableCompression) {
				count = newLength | uncompressedBit;
			} else {
				count = newLength;
			}

			return this;
		}
	} else {
		// Append Integer.MIN_VALUE as a String
		return append("-2147483648"); //$NON-NLS-1$
	}
	/*[ENDIF]*/
```
Would greatly simplify the code.

Reply:
I agree. Fixed in 404b3f0